### PR TITLE
Port argument-non-c-like-enum to Rust

### PR DESCRIPTION
**Pull request by jieyouxu**
Part of #121876.

### CODE REVIEW

File: src/tools/run-make-support/src/cc.rs [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+    //         endif
+    //     endif
+    // endif
+    // ```
+    if is_windows() {
+        if is_msvc() { vec![] } else { vec!["-lstdc++"] }
+    } else {
+        match uname() {
+            n if n.contains("Darwin") => vec!["-lc++"],
+            _ => vec!["-lstdc++"],
```
Comment (riking) [inline src/tools/run-make-support/src/cc.rs:199]: This drops the `vec![]` for FreeBSD, SunOS, OpenBSD that the original had

Comment (jieyouxu) [reply src/tools/run-make-support/src/cc.rs:199]: The original tools.mk had

```makefile
# Extra flags needed to compile a working executable with the standard library
ifdef IS_WINDOWS
ifdef IS_MSVC
	#EXTRACFLAGS := ws2_32.lib userenv.lib advapi32.lib bcrypt.lib ntdll.lib synchronization.lib
else
	#EXTRACFLAGS := -lws2_32 -luserenv -lbcrypt -lntdll -lsynchronization
	EXTRACXXFLAGS := -lstdc++
	#EXTRARSCXXFLAGS := -l static:-bundle=stdc++
endif
else
ifeq ($(UNAME),Darwin)
	#EXTRACFLAGS := -lresolv
	EXTRACXXFLAGS := -lc++
	#EXTRARSCXXFLAGS := -lc++
else
ifeq ($(UNAME),FreeBSD)
	#EXTRACFLAGS := -lm -lpthread -lgcc_s
else
ifeq ($(UNAME),SunOS)
	#EXTRACFLAGS := -lm -lpthread -lposix4 -lsocket -lresolv
else
ifeq ($(UNAME),OpenBSD)
	#EXTRACFLAGS := -lm -lpthread -lc++abi
	#RUSTC := $(RUSTC) -C linker="$(word 1,$(CC:ccache=))"
else
	#EXTRACFLAGS := -lm -lrt -ldl -lpthread
	EXTRACXXFLAGS := -lstdc++
	#EXTRARSCXXFLAGS := -lstdc++
endif
endif
endif
endif
endif
```

It should've been empty `vec![]` for `FreeBSD`, `SunOS` and `OpenBSD` is what you mean right?

Comment (jieyouxu) [reply src/tools/run-make-support/src/cc.rs:199]: Thanks for the catch, I'll send a patch for it

Comment (riking) [reply src/tools/run-make-support/src/cc.rs:199]: Exactly that, you got it

**Pull request merged**